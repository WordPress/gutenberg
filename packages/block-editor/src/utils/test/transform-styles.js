/**
 * Internal dependencies
 */
import transformStyles from '../transform-styles';

describe( 'transformStyles', () => {
	describe( 'selector wrap', () => {
		it( 'should wrap regular selectors', () => {
			const input = `h1 { color: red; }`;
			const output = transformStyles(
				[
					{
						css: input,
					},
				],
				'.my-namespace'
			);

			expect( output ).toMatchSnapshot();
		} );

		it( 'should wrap multiple selectors', () => {
			const input = `h1, h2 { color: red; }`;
			const output = transformStyles(
				[
					{
						css: input,
					},
				],
				'.my-namespace'
			);

			expect( output ).toMatchSnapshot();
		} );

		it( 'should ignore selectors', () => {
			const input = `h1, body { color: red; }`;
			const output = transformStyles(
				[
					{
						css: input,
						ignoredSelectors: [ 'body' ],
					},
				],
				'.my-namespace'
			);

			expect( output ).toMatchSnapshot();
		} );

		it( 'should replace root tags', () => {
			const input = `body, h1 { color: red; }`;
			const output = transformStyles(
				[
					{
						css: input,
					},
				],
				'.my-namespace'
			);

			expect( output ).toMatchSnapshot();
		} );

		it( 'should ignore keyframes', () => {
			const input = `
			@keyframes edit-post__fade-in-animation {
				from {
					opacity: 0;
				}
			}`;
			const output = transformStyles(
				[
					{
						css: input,
					},
				],
				'.my-namespace'
			);

			expect( output ).toMatchSnapshot();
		} );

		it( 'should wrap selectors inside container queries', () => {
			const input = `
			@container (width > 400px) {
				  h1 { color: red; }
			}`;
			const output = transformStyles(
				[
					{
						css: input,
					},
				],
				'.my-namespace'
			);

			expect( output ).toMatchSnapshot();
		} );

		it( 'should ignore font-face selectors', () => {
			const input = `
			@font-face {
				font-family: myFirstFont;
				src: url(sansation_light.woff);
			}`;
			const output = transformStyles(
				[
					{
						css: input,
					},
				],
				'.my-namespace'
			);

			expect( output ).toMatchSnapshot();
		} );

		it( 'should replace :root selectors', () => {
			const input = `
			:root {
				--my-color: #ff0000;
			}`;
			const output = transformStyles(
				[
					{
						css: input,
					},
				],
				'.my-namespace'
			);

			expect( output ).toMatchSnapshot();
		} );

		it( 'should not double wrap selectors', () => {
			const input = ` .my-namespace h1, .red { color: red; }`;

			const output = transformStyles(
				[
					{
						css: input,
					},
				],
				'.my-namespace'
			);

			expect( output ).toMatchSnapshot();
		} );
	} );

	describe( 'URL rewrite', () => {
		it( 'should rewrite relative paths', () => {
			const input = `h1 { background: url(images/test.png); }`;
			const output = transformStyles( [
				{
					css: input,
					baseURL: 'http://wp-site.local/themes/gut/css/',
				},
			] );

			expect( output ).toMatchSnapshot();
		} );

		it( 'should replace complex relative paths', () => {
			const input = `h1 { background: url(../images/test.png); }`;
			const output = transformStyles( [
				{
					css: input,
					baseURL: 'http://wp-site.local/themes/gut/css/',
				},
			] );

			expect( output ).toMatchSnapshot();
		} );

		it( 'should not replace absolute paths', () => {
			const input = `h1 { background: url(/images/test.png); }`;
			const output = transformStyles( [
				{
					css: input,
					baseURL: 'http://wp-site.local/themes/gut/css/',
				},
			] );

			expect( output ).toMatchSnapshot();
		} );

		it( 'should not replace remote paths', () => {
			const input = `h1 { background: url(http://wp.org/images/test.png); }`;
			const output = transformStyles( [
				{
					css: input,
					baseURL: 'http://wp-site.local/themes/gut/css/',
				},
			] );

			expect( output ).toMatchSnapshot();
		} );
	} );
} );
