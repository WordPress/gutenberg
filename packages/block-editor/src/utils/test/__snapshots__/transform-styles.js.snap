// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformStyles URL rewrite should not replace absolute paths 1`] = `
[
  "h1 { background: url(/images/test.png); }",
]
`;

exports[`transformStyles URL rewrite should not replace remote paths 1`] = `
[
  "h1 { background: url(http://wp.org/images/test.png); }",
]
`;

exports[`transformStyles URL rewrite should replace complex relative paths 1`] = `
[
  "h1 { background: url(http://wp-site.local/themes/gut/images/test.png); }",
]
`;

exports[`transformStyles URL rewrite should rewrite relative paths 1`] = `
[
  "h1 { background: url(http://wp-site.local/themes/gut/css/images/test.png); }",
]
`;

exports[`transformStyles selector wrap should ignore font-face selectors 1`] = `
[
  "
			@font-face {
				font-family: myFirstFont;
				src: url(sansation_light.woff);
			}",
]
`;

exports[`transformStyles selector wrap should ignore keyframes 1`] = `
[
  "
			@keyframes edit-post__fade-in-animation {
				from {
					opacity: 0;
				}
			}",
]
`;

exports[`transformStyles selector wrap should ignore selectors 1`] = `
[
  ".my-namespace h1, body { color: red; }",
]
`;

exports[`transformStyles selector wrap should not double wrap selectors 1`] = `
[
  " .my-namespace h1, .my-namespace .red { color: red; }",
]
`;

exports[`transformStyles selector wrap should replace :root selectors 1`] = `
[
  "
			.my-namespace {
				--my-color: #ff0000;
			}",
]
`;

exports[`transformStyles selector wrap should replace root tags 1`] = `
[
  ".my-namespace, .my-namespace h1 { color: red; }",
]
`;

exports[`transformStyles selector wrap should wrap multiple selectors 1`] = `
[
  ".my-namespace h1, .my-namespace h2 { color: red; }",
]
`;

exports[`transformStyles selector wrap should wrap regular selectors 1`] = `
[
  ".my-namespace h1 { color: red; }",
]
`;

exports[`transformStyles selector wrap should wrap selectors inside container queries 1`] = `
[
  "
			@container (width > 400px) {
				  .my-namespace h1 { color: red; }
			}",
]
`;
